DirectProgramming/DPC++FPGA/Tutorials/GettingStarted/fpga_template/src/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+# compilation
+
+set(USER_FLAGS "")
+# use cmake -DUSER_FLAGS=<flags> to set extra flags for general compilation
+
+###############################################################################
+### no changes after here
+###############################################################################
+
+# define the different build targets the generated build script will support.
+set(EMULATOR_TARGET ${TARGET_NAME}.fpga_emu)
+
+# Full-system
+set(BOARD_REPORT_TARGET ${TARGET_NAME}.fpga_report)
+set(BOARD_SIMULATOR_TARGET ${TARGET_NAME}.fpga_sim)
+set(BOARD_TARGET ${TARGET_NAME}.fpga)
+
+# FPGA board selection
+if(DEFINED FPGA_BOARD)
+    message(STATUS "Configuring the design to run on FPGA accelerator board ${FPGA_BOARD}")
+endif()
+
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+    # add qactypes to link command on Windows only
+    set(QACTYPES_LINK "-qactypes")
+endif()
+
+# A DPC++ ahead-of-time (AoT) compile processes the device code in two stages.
+# 1. The "compile" stage compiles the device code to an intermediate
+#    representation (SPIR-V).
+# 2. The "link" stage invokes the compiler's FPGA backend before linking. For
+#    this reason, FPGA backend flags must be passed as link flags in CMake.
+set(EMULATOR_COMPILE_FLAGS "-fintelfpga -Wall ${WIN_FLAG} -qactypes")
+set(EMULATOR_LINK_FLAGS    "-fintelfpga ${QACTYPES_LINK}")
+set(HARDWARE_COMPILE_FLAGS "-fintelfpga -Wall ${WIN_FLAG} -qactypes")
+set(BOARD_LINK_FLAGS       "-fintelfpga -Xstarget=${FPGA_BOARD}   ${USER_HARDWARE_FLAGS} ${QACTYPES_LINK}")
+
+###############################################################################
+### FPGA Emulator
+###############################################################################
+# To compile in a single command:
+#    dpcpp -fintelfpga -DFPGA_EMULATOR fpga_compile.cpp -o fpga_compile.fpga_emu
+# CMake executes:
+#    [compile] dpcpp -fintelfpga -DFPGA_EMULATOR -o fpga_compile.cpp.o -c fpga_compile.cpp
+#    [link]    dpcpp -fintelfpga fpga_compile.cpp.o -o fpga_compile.fpga_emu
+add_executable(${EMULATOR_TARGET} ${SOURCE_FILE})
+add_custom_target(fpga_emu DEPENDS ${EMULATOR_TARGET})
+set_target_properties(${EMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${EMULATOR_COMPILE_FLAGS} -DFPGA_EMULATOR ${USER_FLAGS}")
+set_target_properties(${EMULATOR_TARGET} PROPERTIES LINK_FLAGS "${EMULATOR_LINK_FLAGS} -DFPGA_EMULATOR ${USER_FLAGS}")
+
+###############################################################################
+### Board Report
+###############################################################################
+# To compile manually:
+#   dpcpp -fintelfpga -Xshardware -Xstarget=<FPGA_BOARD> -fsycl-link=early fpga_compile.cpp -o fpga_compile_report.a
+set(BOARD_REPORT_TARGET ${TARGET_NAME}.report.a)
+# The compile output is not an executable, but an intermediate compilation result unique to DPC++.
+add_executable(${BOARD_REPORT_TARGET} ${SOURCE_FILE})
```

Comment:
```suggestion
add_executable(${DEVICE_REPORT_TARGET} ${SOURCE_FILE})
```

Reply:
I will actually just change to `REPORT_TARGET` since we won't distinguish between `device` and `board` in the build script; this distinction will be handled by whatever the user passes with `-DDEVICE=` flag to CMake, e.g.

`cmake .. -DDEVICE=Arria10`

Reply:
removed the word 'BOARD' entirely and just show EMULATOR, REPORT, SIMULATOR, FPGA